src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq.c

```diff
@@ -2918,6 +2948,11 @@ static int xgq_vmr_probe(struct platform_device *pdev)
 		goto done;
 	}
 
+	if (!vmr_check_sc_is_ready(xgq)) {
```

Comment:
seems the sc_ready flag is redundant, could we combine 2951 - 2955 with  2982-2987 together and remove sc_ready local variable. Because we always get the from the vmr_status report which is reported by VMR.